bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/handler/BroadlinkBaseThingHandler.java

```diff
@@ -0,0 +1,302 @@
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.util.HexUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract superclass of all supported Broadlink devices.
+ *
+ * @author John Marshall/Cato Sognen - Initial contribution
+ */
+@NonNullByDefault
+public abstract class BroadlinkBaseThingHandler extends BaseThingHandler implements DeviceRediscoveryListener {
+
+    public final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final String INITIAL_DEVICE_ID = "00000000";
+
+    protected BroadlinkDeviceConfiguration thingConfig = new BroadlinkDeviceConfiguration();
+
+    private @Nullable RetryableSocket socket;
+
+    // Can be injected for test purposes
+    private @Nullable NetworkTrafficObserver networkTrafficObserver;
+
+    private int count;
+    private @Nullable ScheduledFuture<?> refreshHandle;
+    private boolean authenticated = false;
+    // These get handed to us by the device after successful authentication:
+    private byte[] deviceId;
+    private byte[] deviceKey;
+
+    public BroadlinkBaseThingHandler(Thing thing) {
+        super(thing);
+        logger.debug("constructed: resetting deviceKey to '{}', length {}",
+                BroadlinkBindingConstants.BROADLINK_AUTH_KEY, BroadlinkBindingConstants.BROADLINK_AUTH_KEY.length());
+        this.deviceId = HexUtils.hexToBytes(INITIAL_DEVICE_ID);
+        this.deviceKey = HexUtils.hexToBytes(BroadlinkBindingConstants.BROADLINK_AUTH_KEY);
+    }
+
+    // For test purposes
+    void setSocket(RetryableSocket socket) {
+        this.socket = socket;
+    }
+
+    void setNetworkTrafficObserver(NetworkTrafficObserver networkTrafficObserver) {
+        this.networkTrafficObserver = networkTrafficObserver;
+    }
+
+    private boolean hasAuthenticated() {
+        return authenticated;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initializing handler");
+
+        this.thingConfig = getConfigAs(BroadlinkDeviceConfiguration.class);
+        count = (new Random()).nextInt(65535);
+
+        if (this.socket == null) {
+            this.socket = new RetryableSocket(thingConfig, logger);
+        }
+
+        updateItemStatus();
+
+        if (thingConfig.getPollingInterval() != 0) {
+            refreshHandle = scheduler.scheduleWithFixedDelay(new Runnable() {
+
+                @Override
+                public void run() {
+                    updateItemStatus();
+                }
+            }, 1L, thingConfig.getPollingInterval(), TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    @SuppressWarnings("null")
+    public void dispose() {
+        logger.debug("Disposing of {}", getThing().getLabel());
+        if (refreshHandle != null && !refreshHandle.isDone()) {
+            logger.debug("Cancelling refresh task");
+            boolean cancelled = refreshHandle.cancel(true);
+            logger.debug("Cancellation successful: {}", cancelled);
```

Comment:
```suggestion
            logger.debug("Cancellation successful: {}", refreshHandle.cancel(true));
```
Slightly more efficient and an example of how it will still run even if debug is disabled. Not necessary to make this change if you do not want to.

Reply:
I don't like mixing control flow change and logging in one line. It reduces readability of the code imho. Hence, I suggest not to implement this change,